azure-iot-device/azure/iot/device/common/async_adapter.py

```diff
@@ -70,9 +70,9 @@ def wrapping_callback(*args, **kwargs):
                 result = None
 
             if exception:
-                # Do not use exc_info parameter on logger.error.  This casuses pytest to save the traceback which saves stack frames which shows up as a leak
-                logger.error("Callback completed with error {}".format(exception))
-                logger.error(traceback.format_exception_only(type(exception), exception))
+                # Do not use exc_info parameter on logger.* calls.  This casuses pytest to save the traceback which saves stack frames which shows up as a leak
+                logger.info("Callback completed with error {}".format(exception))
```

Comment:
do we return this error ? if so info is fine...asking otherwise

Reply:
Yes, it gets raised when the user awaits the future.  

---
In reply to: [477605792](https://github.com/Azure/azure-iot-sdk-python/pull/649#discussion_r477605792) [](ancestors = 477605792)